services/core/IEEE_2030_5/ieee_2030_5/agent.py

```diff
@@ -0,0 +1,804 @@
+                        point_value = getattr(point_value, "value")
+
+                    if point_value:
+                        self.vip.rpc.call("platform.driver", "set_point", self._device_topic,
+                                          point.point_on_bus, point_value)
+            except TypeError:
+                _log.error(f"Error setting point {point.point_on_bus} to {point_value}")
+            except KeyError:
+                _log.error(f"Error setting point {point.point_on_bus} to {point_value}")
+
+        for point in der_base_points:
+
+            point_value = getattr(default_control.DERControlBase, point.parameter)
+
+            try:
+                if point_value:
+                    if not isinstance(point_value, (float, int, bool)):
+                        point_value = getattr(point_value, "value")
+
+                    if point_value:
+                        self.vip.rpc.call("platform.driver", "set_point", point.point_on_bus,
+                                          point_value)
+            except TypeError:
+                _log.error(f"Error setting point {point.point_on_bus} to {point_value}")
+
+    def _control_event_started(self, sender: m.DERControl):
+        _log.debug(f"{'='*50}Control event started\n{sender}")
+        if not isinstance(sender, m.DERControl):
+            _log.error("Invalid control event passed to event_started")
+            raise ValueError(
+                f"Invalid type passed to event_started {type(sender)} instead of {type(m.DERControl)}"
+            )
+
+        self._current_control = sender
+        der_control: m.DERControl = sender
+        # We override some of the base controls with the event controls
+        der_control_base: m.DERControlBase = None
+        if self._default_der_control is not None and self._default_der_control.DERControlBase is not None:
+            der_control_base = deepcopy(self._default_der_control.DERControlBase)
+
+        if der_control_base:
+            # Overwrite all of the base controls with the controls from the event.
+            for fld in fields(m.DERControlBase):
+                setattr(der_control_base, fld.name, getattr(der_control.DERControlBase, fld.name))
+        else:
+            der_control_base = der_control.DERControlBase
+
+        # Retrieve mapped points that we can report on to the platform driver.
+        # Note this is a DERControlBase prefix in the parameter_type field.
+        der_base_points = list(
+            filter(lambda x: 'DERControlBase' in x.parameter_type, self._mapped_points.values()))
+
+        for point in der_base_points:
+
+            point_value = getattr(der_control_base, point.parameter)
+
+            try:
+                if point_value:
+                    # These are the point types that have a multiplyer assigned to them.
+                    if isinstance(point_value,
+                                  (m.VoltageRMS, m.ApparentPower, m.PowerFactor, m.CurrentRMS,
```

Comment:
Are these points specific to a inverter. Would different der devices have different set of points that would have a multiplier. If so is this just a subset we support?

Reply:
No they aren't they are specific to DERs.

Reply:
So what der is this for? Are point different for say a solar panel vs wind power generator? And does this mean this implementation is for a specific DER type?